src/peep/peep.c

```diff
@@ -3421,7 +3453,761 @@ static void peep_update_fixing(int steps, rct_peep* peep){
+			sprite->vehicle.update_flags &= 0xFDFF;
+			return;
+		}
+
+		sprite = &g_sprite_list[sprite->vehicle.next_vehicle_on_train];
+	}
+}
+
+void loc_6C1114(int steps, rct_peep *peep, rct_ride *ride) {
+
+	if (ride_type_has_flag(ride->type, 0x8008)) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	uint16 ax = ride->station_starts[peep->current_ride_station];
+	if (ax == 0xFFFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+
+	uint8 dl = ride->station_heights[peep->current_ride_station];
+
+
+	uint16 cx = 0;
+	uint8 cl = ax >> 8;
+	uint8 al = ax & 0xFF;
+
+	ax = al << 5;
+	cx = cl << 5;
+
+	// loc_6C1175:
+	rct_map_element *mapElement = map_get_track_element_at(ax, cx, dl);
+	if (mapElement == NULL) {
+		log_error("Couldn't find map_element");
+		return;
+	}
+
+	// loc_6C118A:
+	int ebx = map_element_get_direction(mapElement);
+	ax += 10;
+	cx += 10;
+	uint16 dx = RCT2_ADDRESS(0x992A3C, uint16)[ebx * 2];
+	ax += dx;
+	if (dx == 0) {
+		ax = peep->destination_x;
+	}
+
+	// loc_6C11AC:
+	dx = RCT2_ADDRESS(0x992A3E, uint16)[ebx * 2];
+	cx += dx;
+	if (dx == 0) {
+		cx = peep->destination_y;
+	}
+
+	// loc_6C11C0:
+	peep->destination_x = ax;
+	peep->destination_y = cx;
+	peep->destination_tolerence = 2;
+
+	loc_6C11CC(steps, peep, ride);
+}
+
+static void loc_6C11CC(int steps, rct_peep *peep, rct_ride *ride) {
+	sint16 x, y, tmp_distance;
+
+	invalidate_sprite_2((rct_sprite *) peep);
+	int out = peep_update_action(&x, &y, &tmp_distance, peep);
+	if (out == 0) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	sprite_move(x, y, peep->z, (rct_sprite *) peep);
+	invalidate_sprite_2((rct_sprite *) peep);
+}
+
+void loc_6C11F5(int steps, rct_peep *peep, rct_ride *ride) {
+	// eax: steps
+	// esi: peep
+	// edi: ride
+
+	uint8 direction = peep->var_78 << 3;
+	peep->sprite_direction = direction;
+	peep->action = PEEP_ACTION_STAFF_CHECKBOARD;
+	peep->action_frame = 0;
+	peep->action_sprite_image_offset = 0;
+
+	sub_693B58(peep);
+	invalidate_sprite_2((rct_sprite *) peep);
+
+	loc_6C121F(steps, peep, ride);
+}
+
+static void loc_6C121F(int steps, rct_peep *peep, rct_ride *ride) {
+	sint16 tmp_x, tmp_y, tmp_xy_distance;
+
+	if (peep->action == 0xFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	peep_update_action(&tmp_x, &tmp_y, &tmp_xy_distance, peep);
+}
+
+void loc_6C1239(int steps, rct_peep *peep, rct_ride *ride) {
+	// eax: steps
+	// esi: peep
+	// edi: ride
+
+	if (!ride_type_has_flag(ride->type, 0x8008)) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	uint16 ax = ride->station_starts[peep->current_ride_station];
+
+	if (ax == 0xFFFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	uint8 dl = ride->station_heights[peep->current_ride_station];
+	uint8 dh = peep->current_ride;
+
+	uint16 cx = ax >> 8;
+	ax &= 0x00FF;
+
+	ax = ax << 5;
+	cx = cx << 5;
+
+	loc_6C1288(steps, peep, ride, ax, cx, dh, dl);
+}
+
+void static loc_6C1288(int steps, rct_peep *peep, rct_ride *ride, uint16 ax, uint16 cx, uint8 dh, uint8 dl) {
```

Comment:
I think this function should basically be removed and replaced with track_get_previous. It would require a bit of thought with how to handle it but it would make it mean we don't reimplement get_previous for this one use case.